src/sm/networkmanager/networkmanager.cpp

```diff
@@ -807,4 +876,118 @@ Error NetworkManager::CreateDNSPluginConfig(
     return ErrorEnum::eNone;
 }
 
+Error NetworkManager::RemoveNetworks(const Array<aos::NetworkParameters>& networks)
+{
+    Error err;
+
+    UniqueLock lock(mMutex);
+
+    for (auto it = mNetworkProviders.begin(); it != mNetworkProviders.end();) {
+        auto itNetwork = networks.FindIf([&](const auto& network) { return it->mFirst == network.mNetworkID; });
+        if (itNetwork == networks.end()) {
+            lock.Unlock();
+
+            if (auto errRemoveNetwork = RemoveNetwork(it->mFirst);
+                !errRemoveNetwork.IsNone() && !errRemoveNetwork.Is(ErrorEnum::eNotFound) && err.IsNone()) {
+                err = errRemoveNetwork;
+            }
+
+            lock.Lock();
+
+            if (auto errClearNetwork = ClearNetwork(it->mFirst); !errClearNetwork.IsNone() && err.IsNone()) {
+                err = errClearNetwork;
+            }
+
+            it = mNetworkProviders.Erase(it);
+
+            if (auto errRemoveStorage = mStorage->RemoveNetworkInfo(it->mFirst);
+                !errRemoveStorage.IsNone() && err.IsNone()) {
+                err = errRemoveStorage;
+            }
+        } else {
+            ++it;
+        }
+    }
+
+    return err;
+}
+
+Error NetworkManager::RemoveNetwork(const String& networkID)
+{
+    LOG_DBG() << "Remove network: networkID=" << networkID;
+
+    UniqueLock lock(mMutex);
+
+    auto provider = mNetworkData.Find(networkID);
+    if (provider == mNetworkData.end()) {
+        return ErrorEnum::eNotFound;
+    }
+
+    Error err;
```

Comment:
Move error definition out of lock. And for such a case better to use lock like this:
```c++
{
    LockGuard lock {mMuxte};

   ....
}
```

Reply:
Done